ui/src/user/Login.tsx

```diff
@@ -4,21 +4,31 @@ import TextField from '@material-ui/core/TextField';
 import React, {Component, FormEvent} from 'react';
 import Container from '../common/Container';
 import DefaultPage from '../common/DefaultPage';
+import {CurrentUser} from '../CurrentUser';
 import {observable} from 'mobx';
 import {observer} from 'mobx-react';
-import {inject, Stores} from '../inject';
+import {inject} from '../inject';
+import RegistrationDialog from './Register';
+import axios from 'axios';
+
+type Props = {
+    registration: boolean;
+    currentUser: CurrentUser;
+};
```

Comment:
Use Stores in here and rename registration to showRegister.
```suggestion
type Props = Stores<'currentUser'> & {
    showRegister: boolean;
};
```